### ISSUE TITLE
Type 'typeof import("ember-modifier")' is not a constructor function type.

### ISSUE THREAD

**Issue by kevinkucharczyk** [ember-modifier/ember-modifier#80]
I have recently created a new addon on Ember 3.25.3, and I added Typescript support via ember-cli-typescript at 4.1.0 (typescript version is 4.2.4).

When consuming this addon in our host app (also using typescript), I can't get it to build, because I'm getting TS errors:
```
node_modules/my-addon/modifiers/my-modifier.d.ts:5:43 - error TS2507: Type 'typeof import("ember-modifier")' is not a constructor function type.

5 export default class MyModifier extends Modifier {
```

The addon has run `ember ts:precompile`, which generates the `modifiers/my-modifier.d.ts` file where the warning appears.

I'm not sure how to fix this error, is there something wrong with the TS declarations in this addon?

**Comment by kevinkucharczyk** [comment]
Hi @chriskrycho, thanks for pulling me in again! I checked, and it looks like I'm still getting that error. I'm really not sure what could be causing it.
I wouldn't be surprised if it was something that's not configured well in our addons/host app, but then again I'm not sure what to look for. I was hoping the [ember-cli-typescript docs](https://docs.ember-cli-typescript.com/ts/with-addons) would yield some answers, but we should be fine according to that.

**Comment by chriskrycho** [comment]
Thanks for following up. Can you share the full contents of that `.d.ts` file (or at least the imports and exports, even if you have to elide the body)? Also, can you show how you’re consuming it?

**Comment by kevinkucharczyk** [comment]
Sure, adding it below. This is `modifiers/draggable.d.ts`, which is generated by the `ts:precompile` command from `addon/modifiers/draggable.ts` in our project (currently a private repo, so I can't link to the whole source, sorry!).
The `//  @ts-ignore` line I added manually, and is where the TS error happens

```
import Modifier from 'ember-modifier';
import { Sensor } from '../sensors';
import { Plugin } from '../plugins';
import { FlyDragEvent } from '../utils/events';
import FlyDragDropManager from '../services/fly-drag-drop-manager';
import { DraggableModifierOptions } from '../';
interface DraggableModifierArgs {
    positional: [];
    named: DraggableModifierOptions;
}
// @ts-ignore
export default class DraggableModifier extends Modifier<DraggableModifierArgs> {
    flyDragDropManager: FlyDragDropManager;
    id: string;
    options: DraggableModifierOptions;
    initialOffset: {
        x: number;
        y: number;
    };
    sensors: Sensor[];
    plugins: Plugin[];
    didReceiveArguments(): void;
    didInstall(): void;
    willDestroy(): void;
    onBeforeDragStart(event: FlyDragEvent): void;
    onDragStart(event: FlyDragEvent): void;
    onDragMove(event: FlyDragEvent): void;
    onDragEnd(event: FlyDragEvent): void;
}
export {};
```

**Comment by chriskrycho** [comment]
Huh. That’s *very* odd, and doesn’t actually make any sense to me. What version(s) of `ember-modifier` do you have installed? And what do the type defs for `ember-modifier` look like on disk (at `node_modules/ember-modifier`?

(I am also happy to hop on a call and pair through this sometime later this week. Currently a bit sick 🤧 but once I’m back on my feet, I’d love to help get to the bottom of this.)

**Comment by kevinkucharczyk** [comment]
@chriskrycho I found the culprit and I am SO SORRY. Turns out our host app was overwriting the type declarations from `ember-modifier` - right under my nose in `types/ember-modifier.d.ts`. Never crossed my mind to check there  😱
We had that in place presumably because we started using `ember-modifier` before TS support was added, and this is all we typed:
```
declare module 'ember-modifier' {
  export function modifier(...args: any[]): any;
}
```

Deleting that file, obviously, fixes the problem. Sorry it took me so long to figure this out, and thanks for your help!
